test-proxy/src/main/java/com/google/cloud/bigtable/testproxy/CbtTestProxy.java

```diff
@@ -0,0 +1,768 @@
+      } catch (RuntimeException e) {
+        // If client encounters problem, fail the whole operation.
+        responseObserver.onNext(
+            RowResult.newBuilder()
+                .setStatus(
+                    com.google.rpc.Status.newBuilder()
+                        .setCode(Code.INTERNAL.getNumber())
+                        .setMessage(e.getMessage())
+                        .build())
+                .build());
+        responseObserver.onCompleted();
+        return;
+      }
+    } else {
+      logger.info(
+          String.format(
+              "readModifyWriteRow() did not find row: %s", request.getRequest().getRowKey()));
+      responseObserver.onNext(
+          RowResult.newBuilder().setStatus(com.google.rpc.Status.getDefaultInstance()).build());
+    }
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public synchronized void close() {
+    Iterator<Map.Entry<String, CbtClient>> it = idClientMap.entrySet().iterator();
+    while (it.hasNext()) {
+      Map.Entry<String, CbtClient> entry = it.next();
+      entry.getValue().dataClient().close();
+      it.remove();
+    }
+  }
+
+  private static String extractTableIdFromTableName(String fullTableName)
+      throws IllegalArgumentException {
+    Matcher matcher = tablePattern.matcher(fullTableName);
+    if (!matcher.matches()) {
+      throw new IllegalArgumentException("Invalid table name: " + fullTableName);
+    }
+    return matcher.group(3);
+  }
+
+  private InstantiatingGrpcChannelProvider getTransportChannel() throws IOException {
+    if (!encrypted) {
+      return EnhancedBigtableStubSettings.defaultGrpcTransportProviderBuilder()
+          .setChannelConfigurator(ManagedChannelBuilder::usePlaintext)
+          .build();
+    }
+
+    if (rootCerts == null) {
+      return EnhancedBigtableStubSettings.defaultGrpcTransportProviderBuilder().build();
+    }
+
+    final SslContext secureContext =
+        GrpcSslContexts.forClient()
+            .trustManager(new ByteArrayInputStream(rootCerts.getBytes(UTF_8)))
+            .build();
+    return EnhancedBigtableStubSettings.defaultGrpcTransportProviderBuilder()
+        .setChannelConfigurator(
+            new ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder>() {
+              @Override
+              public ManagedChannelBuilder apply(ManagedChannelBuilder input) {
+                NettyChannelBuilder channelBuilder = (NettyChannelBuilder) input;
+                channelBuilder.sslContext(secureContext).overrideAuthority(sslTarget);
+                return channelBuilder;
+              }
+            })
+        .build();
+  }
+
+  private CredentialsProvider getCredentialsProvider() throws IOException {
+    if (credential == null) {
+      return NoCredentialsProvider.create();
+    }
+
+    final ServiceAccountJwtAccessCredentials creds =
+        ServiceAccountJwtAccessCredentials.fromStream(
+            new ByteArrayInputStream(credential.getBytes(UTF_8)));
```

Comment:
why are you hardcoding jwt creds? I think GoogleCredentials.fromStream() is more appropriate